token/program-2022-test/tests/delegate.rs

```diff
@@ -114,26 +129,57 @@ async fn run_basic(
     // transfer is ok
     if transfer_mode == TransferMode::All {
         token
-            .transfer_unchecked(&alice_account, &bob_account, &bob, 1)
+            .transfer(
+                &alice_account,
+                &bob_account,
+                &bob.pubkey(),
+                1,
+                None,
+                &vec![&bob],
+            )
             .await
             .unwrap();
     }
 
     token
-        .transfer_checked(&alice_account, &bob_account, &bob, 1, decimals)
+        .transfer(
+            &alice_account,
+            &bob_account,
+            &bob.pubkey(),
+            1,
+            Some(decimals),
+            &vec![&bob],
+        )
         .await
         .unwrap();
 
     // burn is ok
-    token.burn(&alice_account, &bob, 1).await.unwrap();
     token
-        .burn_checked(&alice_account, &bob, 1, decimals)
+        .burn(&alice_account, &bob.pubkey(), 1, None, &vec![&bob])
+        .await
+        .unwrap();
+    token
+        .burn(
+            &alice_account,
+            &bob.pubkey(),
+            1,
+            Some(decimals),
+            &vec![&bob],
+        )
         .await
         .unwrap();
 
     // wrong signer
+    let keypair = &Keypair::new();
```

Comment:
TIL: I didn't know you could directly grab a reference to something during creation like this